AndroidPubSub/AuthenticatedPubsub.md

```diff
@@ -0,0 +1,86 @@
+
+    ```json
+    {
+        "Version": "2012-10-17",
+        "Statement": [
+            {
+                "Effect": "Allow",
+                "Action": [
+                    "iot:AttachPrincipalPolicy",
+                    "iot:Connect",
+                    "iot:Publish",
+                    "iot:Subscribe",
+                    "iot:Receive"
+                ],
+                "Resource": [
+                    "*"
+                ]
+            }
+        ]
+    }
+    ```
+
+1. In the app, get the credentials provider
+
+    ```java
+    credentialsProvider =
+                   IdentityManager.getDefaultIdentityManager().getCredentialsProvider();
+    ```
+
+1. In the app, after the user logs in using the sign-in UI, get `CognitoIdentity` from `credentialsProvider` and use it to attach the principal policy to the `IotAndroidClient`. Create the IoT policy as specified in step 4 of the [Using the sample](https://github.com/awslabs/aws-sdk-android-samples/tree/master/AndroidPubSub#using-the-sample) section above.
+
+    ```java
+    // Setup the logins map
+    IdentityManager identityManager = IdentityManager.getDefaultIdentityManager();
+    Map<String, String> logins = new HashMap();
+    IdentityProvider currentIdentityProvider = identityManager.getCurrentIdentityProvider();
+    String token = currentIdentityProvider.getToken();
+    String providerKey = currentIdentityProvider.getCognitoLoginKey();
+    logins.put(providerKey, token);
+
+    AmazonCognitoIdentity cognitoIdentity;
+    GetIdRequest getIdReq;
+    GetIdResult getIdRes;
+    cognitoIdentity = new AmazonCognitoIdentityClient(credentialsProvider);
+    getIdReq = new GetIdRequest();
+    getIdReq.setLogins(logins);
+    getIdReq.setIdentityPoolId(COGNITO_POOL_ID); //Identity pool Id generated by amplify cli
+    new Thread(new Runnable() {
+        @Override
+        public void run() {
+        getIdRes = cognitoIdentity.getId(getIdReq);
+        AttachPrincipalPolicyRequest attachPolicyReq = new AttachPrincipalPolicyRequest();
+        attachPolicyReq.setPolicyName(AWS_IOT_POLICY_NAME);
+        attachPolicyReq.setPrincipal(getIdRes.getIdentityId());
+        AWSIotClient mIotAndroidClient = new AWSIotClient(credentialsProvider);
+        mIotAndroidClient.setRegion(Region.getRegion(MY_REGION));
+        mIotAndroidClient.attachPrincipalPolicy(attachPolicyReq);
+        }
+    }).start();
+    ```
+
+    An Amazon Cognito authenticated user needs two policies to access AWS IoT. The first policy is attached to the role of the authenticated pool to authenticate and authorize the Cognito user to communicate with AWS IoT. The second policy is attached to the authenticated Cognito user ID principal for fine-grained permissions.
```

Comment:
This section feels like an introduction that should be followed by examples of implementation. Are the two policies created in this code, in the inline policy JSON in the section above this one, or via some other mechanism?